subscription-service/src/main/kotlin/com/egm/stellio/subscription/config/ScheduledNotification.kt

```diff
@@ -0,0 +1,125 @@
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
+
+@Component
+class ScheduledNotification(
+    private val subscriptionService: SubscriptionService,
+    private val notificationService: NotificationService
+) {
+
+    @Autowired
+    private val webClient: WebClient? = null
+
+    /**
+     * Every seconds, the function cheks the notification to send
+     *
+     */
+    @Scheduled(fixedRate = 1000)
+    suspend fun scheduledSendNotification() {
+        val litSubscription = subscriptionService.getAllSubscriptions()
+        for (subscription in litSubscription) {
+            for (entityParam in subscription.entities) {
+                if (entityParam.id != null)
+                    sendNotification(getEntity(extractParam(entityParam, subscription.q)), subscription)
+                else sendNotification(getEntitiesByType(extractParam(entityParam, subscription.q)), subscription)
+            }
+        }
+    }
+
+    fun extractParam(listEntityInfo: Set<EntityInfo>, q: String?): List<String> {
+        val listParam = ArrayList<String>()
+        for (entityInfo in listEntityInfo) listParam.add(extractParam(entityInfo, q))
+        return listParam
+    }
+
+    /**
+     * Function for extram param of subscription
+     * Param are use to have entity when we call 'entityService'
+     *
+     * @return list of string
+     */
+    fun extractParam(entityInfo: EntityInfo, q: String?): String {
+        val param = java.lang.StringBuilder()
+        if (entityInfo.id != null) {
+            param.append("/${entityInfo.id}")
+            if (q != null) param.append("?q=$q")
+        } else {
+            param.append("?type=${entityInfo.type}")
+            if (entityInfo.idPattern != null) param.append("&idPattern=${entityInfo.idPattern}")
+            if (q != null) param.append("&q=$q")
+        }
+        return param.toString()
+    }
+
+    /**
+     * For each JsonLdEntity we call sendNotification
+     *
+     *  @return List<Mono<Triple<Subscription, Notification, Boolean>>>
+     */
+    fun sendNotification(listEntity: List<JsonLdEntity>, subscription: Subscription):
+        List<Mono<Triple<Subscription, Notification, Boolean>>> {
+        val listResultSendNotification = ArrayList<Mono<Triple<Subscription, Notification, Boolean>>>()
+        for (entity in listEntity) listResultSendNotification.add(sendNotification(entity, subscription))
+        // listEntity.forEach { listResultSendNotification.add(sendNotification(it, subscription)) }
+        return listResultSendNotification
+    }
+
+    /**
+     * Function to send notificatio
+     * We call 'callSubscriber'
+     *
+     * @return Mono<Triple<Subscription, Notification, Boolean>>
+     */
+    fun sendNotification(entity: JsonLdEntity, subscription: Subscription):
```

Comment:
if you don't subscribe to the result (or use one the `awaitXXX` function in the calling code), the `callSubscriber(...)` will never be executed since it returns a Mono (like a promise of a result)

Reply:
I add awaitFirst